reactioncommerce/reaction-logger
# fix: ensure compatibility with non-Babel Node imports

When using this package on latest versions of Node without Babel, including with ES module support turned on, it doesn't import quite right.

Previously, you had to do this workaround:

```js
import logger from "@reactioncommerce/logger";
const Logger = logger.default;
```

Or without ESM:

```js
const Logger = require("@reactioncommerce/logger").default;
```

The fix was two parts:
1. Add `babel-plugin-add-module-exports` Babel plugin package
2. Remove the named exports, since that plugin only works if there are no named exports. I set `bunyan` and `bunyanFormat` properties on the default `Logger` export instead, which is icky but has the effect of keeping everything working the same if you are using Babel. (You can still `import { bunyan } from "@reactioncommerce/logger";`).

## Test
1. Verify `import Logger from "@reactioncommerce/logger";` now works in a non-Babel Node project with ES modules enabled.
1. Verify `const Logger = require("@reactioncommerce/logger");` now works in a non-Babel Node project with or without ES modules enabled.
1. Verify `import Logger from "@reactioncommerce/logger";` still works in a Babel project.
1. Verify `import { bunyan } from "@reactioncommerce/logger";` still works in a Babel project.
